### PR TITLE
Feature/stats monitor microservice

### PR DESCRIPTION
**Pull request by SasukeVapes**
Added Microservice with Regular DAL and DALExternal

### CODE REVIEW

**Review by blackberts — APPROVED** [review]

File: StatsMonitor/StatsMonitor.DAL/DI/DependencyInjection.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using StatsMonitor.DAL.Data;
+using StatsMonitor.DAL.Interfaces;
+using StatsMonitor.DAL.Repositories;
+
+namespace StatsMonitor.DAL.DI
+{
+    public static class DependencyInjection
+    {
+        public static void AddDependenciesDalLayer(this IServiceCollection services, IConfiguration config)
```
Comment (Glipchik) [inline StatsMonitor/StatsMonitor.DAL/DI/DependencyInjection.cs:12]: Either DALLayer or DAL, not DalLayer

**Review by Glipchik — APPROVED** [review]

File: StatsMonitor/StatsMonitor.BLL/Jobs/UpdateMultipleStatsJob.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            var pokemonOwners = await _pokemonOwnerExternalService.GetPokemonOwners(token);
+            var pokemons = await _pokemonRepository.GetPokemons(token);
+            //Get ALL PokemonOwner Entities, specifically we need Health, Hunger, Happiness from main API
+            foreach (var pOw in pokemonOwners)
+            {
+                var pokemon = pokemons.FirstOrDefault(p => pOw.PokemonId == p.Id);
+                if (pOw.Hunger > 0)
+                { pOw.Hunger -= pokemon.HungerRate; }
+                if (pOw.Hunger < 0.5m && pOw.Hunger > 0)
```
Comment (GodlyCat) [inline StatsMonitor/StatsMonitor.BLL/Jobs/UpdateMultipleStatsJob.cs:29]: move 0, 0.5m , 0.25m to constants

File: StatsMonitor/StatsMonitor.BLL/Jobs/UpdateStatsJob.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        CancellationToken token;
+        public UpdateStatsJob(IPokemonOwnerExternalService pokemonOwnerExternalService, IPokemonRepository pokemonRepository)
+        {
+            _pokemonOwnerExternalService = pokemonOwnerExternalService;
+            _pokemonRepository = pokemonRepository;
+            token = source.Token;
+        }
+        public async Task Execute(IJobExecutionContext context)
+        {
+            var pokemonOwners = await _pokemonOwnerExternalService.GetPokemonOwners(token);
```
Comment (GodlyCat) [inline StatsMonitor/StatsMonitor.BLL/Jobs/UpdateStatsJob.cs:21]: Why you have 2 similar jobs ?

File: StatsMonitor/StatsMonitor.BLL/Jobs/UpdateStatsJob.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        private readonly IPokemonRepository _pokemonRepository;
+        CancellationTokenSource source = new CancellationTokenSource();
+        CancellationToken token;
+        public UpdateStatsJob(IPokemonOwnerExternalService pokemonOwnerExternalService, IPokemonRepository pokemonRepository)
+        {
+            _pokemonOwnerExternalService = pokemonOwnerExternalService;
+            _pokemonRepository = pokemonRepository;
+            token = source.Token;
+        }
+        public async Task Execute(IJobExecutionContext context)
```
Comment (GodlyCat) [inline StatsMonitor/StatsMonitor.BLL/Jobs/UpdateStatsJob.cs:19]: Move this part to service and invoke method from service in job

File: StatsMonitor/StatsMonitor.BLL/Services/PokemonService.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
+﻿using AutoMapper;
+using StatsMonitor.BLL.Abstractions;
+using StatsMonitor.BLL.Models;
+using StatsMonitor.DAL.Entities;
+using StatsMonitor.DAL.Interfaces;
+
+namespace StatsMonitor.BLL.Services
+{
+    public class PokemonService : IPokemonService
```
Comment (GodlyCat) [inline StatsMonitor/StatsMonitor.BLL/Services/PokemonService.cs:9]: Are you sure about count of methods ? (currently needed only one)

File: StatsMonitor/StatsMonitor.BLL/Services/PokemonOwnerService.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            _mapper = mapper;
+        }
+        public async Task<IEnumerable<PokemonOwner>> GetPokemonOwners(CancellationToken cancellationToken)
+        {
+            var pokemonOwners = await _pokemonOwnerExternalService.GetPokemonOwners(cancellationToken);
+
+            return _mapper.Map<List<PokemonOwner>>(pokemonOwners);
+        }
+
+        public async Task<PokemonOwner> UpdatePokemonOwner(int ownerId, int pokemonId, PokemonOwner updatedPokemonOwner, CancellationToken cancellationToken)
```
Comment (GodlyCat) [inline StatsMonitor/StatsMonitor.BLL/Services/PokemonOwnerService.cs:25]: Why update use both ids instead of PoW id ?

**Review by GodlyCat — CHANGES_REQUESTED** [review]

**Review by GodlyCat — APPROVED** [review]

**Pull request merged**